From d7bcb2c6ce702651b6c8c5f90bb0aec01e352b4f Mon Sep 17 00:00:00 2001
From: Zoharit Hadad <zoharithadad@microsoft.com>
Date: Mon, 12 May 2025 11:31:18 +0000
Subject: Avoid gcp content length error

---
 registry/remote/repository.go | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/registry/remote/repository.go b/registry/remote/repository.go
index fed993d..2fd37b9 100644
--- a/registry/remote/repository.go
+++ b/registry/remote/repository.go
@@ -1096,9 +1096,6 @@ func (s *manifestStore) Fetch(ctx context.Context, target ocispec.Descriptor) (r
 	if mediaType != target.MediaType {
 		return nil, fmt.Errorf("%s %q: mismatch response Content-Type %q: expect %q", resp.Request.Method, resp.Request.URL, mediaType, target.MediaType)
 	}
-	if size := resp.ContentLength; size != -1 && size != target.Size {
-		return nil, fmt.Errorf("%s %q: mismatch Content-Length", resp.Request.Method, resp.Request.URL)
-	}
 	if err := verifyContentDigest(resp, target.Digest); err != nil {
 		return nil, err
 	}
-- 
2.34.1

